cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceMetricsWriter.java

```diff
@@ -95,7 +95,15 @@ private void reportMapredStats() throws IOException, InterruptedException {
     int memoryPerMapper = jobConf.getConfiguration().getInt(Job.MAP_MEMORY_MB, Job.DEFAULT_MAP_MEMORY_MB);
     int memoryPerReducer = jobConf.getConfiguration().getInt(Job.REDUCE_MEMORY_MB, Job.DEFAULT_REDUCE_MEMORY_MB);
 
+    Counters counters = jobConf.getCounters();
```

Comment:
If this is not local (rpc call?), then just use the same instance that's used in `reportSystemStats` method.